femtools/Detector_Data_Types.F90

```diff
@@ -116,11 +116,8 @@ module detector_data_types
      integer :: num_vfields = 0   ! Total number of vector fields across all phases
 
      !! I/O parameters
-     logical :: binary_output = .false.
      logical :: write_nan_outside = .false.
-     integer :: output_unit = 0          ! Assumed non-opened as long this is 0
-     integer :: mpi_fh = 0               ! MPI filehandle
-     integer :: mpi_write_count = 0      ! Offset in MPI file
+     integer(kind=8) :: h5_id = -1 ! H5hut output identifier
```

Comment:
I think you asked about this one (is it ok to hard-code `kind=8`?): if you wanted to guarantee some the range you would use `selected_int_kind`. Here, we're actually trying to be compatible with a `int64_t` in C so in principle the correct thing would be to use the `iso_c_binding` module and use `kind=c_int64_t`. However, since we are using the fortran wrappers provided with h5hut (which probably predate the `iso_c_binding` stuff) with explicit interfaces that hard-code integer*8 - I think the correct thing to do is what you did and simply use `kind=8` ourselves. If that fails on some obscure compiler then that's a failure in h5hut really.

Reply:
Okay, good to have confirmation of this.